### ISSUE TITLE
running multiple mympirun commands at the same time doesn't always work

### ISSUE THREAD

**Issue by boegel** [hpcugent/vsc-mympirun#95]
The directory created in `$HOME/.mympirun` is quite prone to name clashes when multiple `mympirun` commands are being run at the same time.

The directory name should be salted with a couple of random characters and/or the (parent?) PID.

```
   File 
"/gpfs/scratch/apps/develop/software/Python/2.7.10-intel-2015b/lib/python2.7/os.py", 
line 157, in makedirs
     mkdir(name, mode)
)
2016-11-29 09:36:42,028 WARNING mympirun.IntelHydraMPI_PBS MainThread  
make_mympirun_dir: failed to make job dir 
/user/home/gent/vsc400/vsc40000/.mympirun/756811.master-moab.muk.os_20161129_093641 
([Errno 17] File exists: 
'/user/home/gent/vsc400/vsc40000/.mympirun/756811.master-moab.muk.os_20161129_093641'
   File 
"/gpfs/scratch/apps/develop/software/vsc-mympirun/3.4.2-intel-2015b-Python-2.7.10-vsc-base-2.4.2/lib
/python2.7/site-packages/vsc_mympirun-3.4.2-py2.7.egg/vsc/mympirun/mpi/mpi.py",

line 579, in make_mympirundir
     os.makedirs(destdir)
```

**Comment by boegel** [comment]
example traceback in description was obtained via use of GNU parallel

a workaround is to specify an alternative directory other than `$HOME` (and ensure it's unique) via `--basepath`:

```
seq 0 3 | parallel -j 4 "mkdir -p /tmp/$USER/$$/mympirun_${1} && mympirun --basepath /tmp/$USER/$$/mympirun_${1}/ -h 4 echo ${1}"
```

**Comment by stdweird** [comment]
why is the mkdir required? also, that command will mess up pinning. myscoop has proper support for running mympirun with pinned tasks, not sure what it does with the basepath option

**Comment by boegel** [comment]
@stdweird `mympirun` now requires that the directory used for `--basepath` already exists

It's just a workaround until we have an updated `mympirun` in place where this is no longer an issue.

**Comment by boegel** [comment]
fixed in #103
